controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -42,11 +46,21 @@ const InitDiscoveryTopicData = async (req, res) => {
     });
     let suggestedTopics = topicWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedTopics,
-      page: page + 1
+      page: page + 1,
+      total_page: totalData > 0 ? Math.ceil(totalData / limit) : 0
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `totalDataQuery` does not seem to have any replacements, but you are passing `userId` as a replacement in the query execution. This could lead to confusion and potential bugs if the query is updated in the future to include replacements. If `userId` is not needed for this query, it should be removed from the replacements.
```diff
-    let totalData = await sequelize.query(totalDataQuery, {
-      type: QueryTypes.SELECT,
-      replacements: {
-        userId
-      },
-      raw: true
-    });
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      raw: true
+    });
```
Also, error handling seems to be missing for the database query. It's important to handle errors that may occur during the execution of the query to prevent unexpected behavior or crashes. Consider wrapping the query in a try-catch block or use promise-based error handling.
```diff
+    try {
-    let totalData = await sequelize.query(totalDataQuery, {
+      let totalData = await sequelize.query(totalDataQuery, {
+        type: QueryTypes.SELECT,
+        raw: true
+      });
+    } catch (error) {
+      console.error('Error executing totalDataQuery:', error);
+      return res.status(500).json({
+        success: false,
+        message: 'Internal server error',
+      });
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->